lib/active_model/serializer/adapter/json_api.rb

```diff
@@ -37,6 +44,27 @@ def serializable_hash(options = {})
 
         private
 
+        def remove_duplicates(hash)
+          if @options[:prevent_duplicates] && hash[:included] && hash[:data].is_a?(Array)
```

Comment:
Here we only remove duplicates if `hash[:data]` is an array because if data is a single object and it is also in included, then it means that the object has a self-reference. Correct me if I'm wrong please.